backend/src/authentication/session.rs

```diff
@@ -0,0 +1,195 @@
+        let session_key = create_new_session_key();
+        let expires_at = get_expires_at(life_time)? as i64;
+        let created_at = get_current_time()? as i64;
+
+        Ok(Self {
+            session_key,
+            user_id,
+            expires_at,
+            created_at,
+        })
+    }
+
+    /// Get the session user id
+    pub(super) fn user_id(&self) -> u32 {
+        self.user_id
+    }
+
+    /// Get the session key
+    pub(super) fn session_key(&self) -> &str {
+        &self.session_key
+    }
+
+    /// Get a cookie containing this session key
+    pub(super) fn get_cookie(&self) -> Cookie<'static> {
+        CookieBuilder::new(SESSION_COOKIE_NAME, self.session_key.clone())
+            .max_age(cookie::time::Duration::seconds(
+                SESSION_LIFE_TIME.as_secs() as i64
+            ))
+            .build()
+    }
+}
+
+/// Sessions repository
+pub struct Sessions(SqlitePool);
+
+impl Sessions {
+    pub fn new(pool: SqlitePool) -> Self {
+        Self(pool)
+    }
+
+    /// Create a new session, note this converts any i64 timestamps to i64
+    pub(super) async fn create(
+        &self,
+        user_id: u32,
+        life_time: Duration,
+    ) -> Result<Session, AuthenticationError> {
+        let session = Session::new(user_id, life_time)?;
+
+        let saved_session = sqlx::query_as!(
+            Session,
+            r#"INSERT INTO sessions (session_key, user_id, expires_at, created_at)
+            VALUES (?, ?, ?, ?)
+            RETURNING
+                session_key,
+                user_id as "user_id: u32",
+                expires_at,
+                created_at
+            "#,
+            session.session_key,
+            session.user_id,
+            session.expires_at,
+            session.created_at
+        )
+        .fetch_one(&self.0)
+        .await?;
+
+        Ok(saved_session)
+    }
+
+    /// Get a session by its key
+    pub(super) async fn get_by_key(
+        &self,
+        session_key: &str,
+    ) -> Result<Option<Session>, AuthenticationError> {
+        let session: Option<Session> = sqlx::query_as!(
+            Session,
+            r#"
+            SELECT
+                session_key,
+                user_id as "user_id: u32",
+                expires_at,
+                created_at
+            FROM sessions WHERE session_key = ?
+            "#,
+            session_key
+        )
+        .fetch_optional(&self.0)
+        .await?;
+
+        Ok(session)
+    }
+
+    /// Delete a session by its key
+    pub async fn delete(&self, session_key: &str) -> Result<(), AuthenticationError> {
+        sqlx::query!("DELETE FROM sessions WHERE session_key = ?", session_key)
+            .execute(&self.0)
+            .await?;
+
+        Ok(())
+    }
+
+    /// Delete all sessions that have expired
+    pub async fn delete_old_sessions(&self) -> Result<(), AuthenticationError> {
```

Comment:
```suggestion
    pub async fn delete_expired_sessions(&self) -> Result<(), AuthenticationError> {
```
Old session is a bit ambiguous in my opinion, because it can also be the previous session of a user that might still be valid. So using the word expired is more specific.